src/main/groovy/edu/oregonstate/mist/musicapi/resources/AlbumResource.groovy

```diff
@@ -0,0 +1,268 @@
+package edu.oregonstate.mist.musicapi.resources
+
+import edu.oregonstate.mist.api.AuthenticatedUser
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.musicapi.core.Album
+import io.dropwizard.auth.Auth
+import java.util.regex.Pattern
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.PUT
+import javax.ws.rs.DELETE
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.Consumes
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.Response.ResponseBuilder
+import org.skife.jdbi.v2.DBI
+import org.skife.jdbi.v2.Handle
+import org.skife.jdbi.v2.util.IntegerMapper
+
+@Path('/album')
+@groovy.transform.TypeChecked
+class AlbumResource extends Resource {
+    private DBI dbi
+    private Pattern releaseDatePattern = Pattern.compile(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/) // YYYY-MM-DD
+
+    public AlbumResource(DBI dbi) {
```

Comment:
Repeating two lines of boilerplate to declare and initialize a variable does not a DRY violation make.